### PR TITLE
Adding pending test for #269

### PR DESCRIPTION
**Pull request by romanowski**
Zinc is undercompiling in this case. Reported as #269 

For certain reasons this time I have to paste the below disclaimer:
THIS PROGRAM IS SUBJECT TO THE TERMS OF THE BSD 3-CLAUSE LICENSE.

THE FOLLOWING DISCLAIMER APPLIES TO ALL SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE:
THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA,
OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING
IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR
ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.
ONLY THE SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE, IF ANY, THAT ARE ATTACHED TO (OR OTHERWISE ACCOMPANY) THIS SUBMISSION (AND ORDINARY
COURSE CONTRIBUTIONS OF FUTURES PATCHES THERETO) ARE TO BE CONSIDERED A CONTRIBUTION. NO OTHER SOFTWARE CODE OR MATERIALS ARE A CONTRIBUTION.

### CODE REVIEW

File: zinc/src/sbt-test/source-dependencies/fuzzyTypes/pending [hunk 1]
```diff
@@ -0,0 +1,6 @@
+> compile
+> compile
```
Comment (dwijnand) [inline zinc/src/sbt-test/source-dependencies/fuzzyTypes/pending]: why compile twice?

Comment (romanowski) [reply zinc/src/sbt-test/source-dependencies/fuzzyTypes/pending]: yeah, leftover from testing :)

File: zinc/src/sbt-test/source-dependencies/fuzzyTypes/pending [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+$ copy-file changed/FactoryProvider.scala src/main/scala/foo/FactoryProvider.scala
+-> compile
+> clean
+-> compile
```
Comment (dwijnand) [inline zinc/src/sbt-test/source-dependencies/fuzzyTypes/pending]: If Zinc is under-compiling shouldn't this be a success after a clean?

**Comment by jvican** [comment]
Will merge this as soon as CI passes because I'll need to rebase for #272.

**Comment by eed3si9n** [comment]
Is "fuzzy type" an actual term in type theory here?
The term fuzzy reminds me only of [fuzzy logic](https://en.wikipedia.org/wiki/Fuzzy_logic) that was really popular among the Japanese appliance makers in the 90s.

**Comment by jvican** [comment]
@eed3si9n Yeah, actually it sounds weird to me too. Do you think `scalac-type-expansion` would be more correct?

**Comment by eed3si9n** [comment]
Not sure what was the problematic part, but I'm guessing it'd be something more boring like type-projection, parameterized-type, type-reference, as-seen-from-c etc.

**Comment by jvican** [comment]
The root of the problem is scalac doing type expansions that we don't track, so I would go for `expanded-type-projection`.

**Comment by eed3si9n** [comment]
Sounds good to me.

**Review by eed3si9n — APPROVED** [review]

**Review by jvican — APPROVED** [review]
LGTM, thanks @romanowski.

**Pull request merged**